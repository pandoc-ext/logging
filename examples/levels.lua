local logging = require 'logging'

if #arg > 0 then
    logging.setloglevel(tonumber(arg[1]) or 0)
end

logging.temp('loglevel is', logging.loglevel, '(this is always output)')
logging.error('this is output if loglevel >= -1')
logging.warning('this is output if loglevel >= 0')
logging.info('this is output if loglevel >= 1')
logging.debug('this is output if loglevel >= 2')
logging.debug2('this is output if loglevel >= 3')

logging.temp('args', 'are', 'automatically', 'separated', 'with',
             'spaces and a newline is added if necessary')
logging.temp('args can be of any type, e.g.', {maps='like', this='one'},
             'or', {'lists', 'like', 'this', 'one'})
logging.temp('if an arg is too long it will be output on multiple lines',
             {'aardvark', 'bison', 'camel', 'dingo', 'echnidna'}, 'like this')
