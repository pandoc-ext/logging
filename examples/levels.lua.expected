(#) loglevel is 0 (this is always output)
(E) this is output if loglevel >= -1
(W) this is output if loglevel >= 0
(#) args are automatically separated with spaces and a newline is added if necessary
(#) args can be of any type, e.g. {maps: "like", this: "one"} or {[1] "lists", [2] "like", [3] "this", [4] "one"}
(#) if an arg is too long it will be output on multiple lines {
  [1] "aardvark"
  [2] "bison"
  [3] "camel"
  [4] "dingo"
  [5] "echnidna"
} like this
